plugins/flashrom/fu-plugin-flashrom.c

```diff
@@ -184,19 +206,58 @@ fu_plugin_update (FuPlugin *plugin,
 	if (!fu_common_set_contents_bytes (firmware_fn, blob_fw, error))
 		return FALSE;
 
-	/* use flashrom to write image */
-	argv[4] = firmware_fn;
-	if (!fu_common_spawn_sync ((const gchar * const *) argv,
-				   fu_plugin_flashrom_write_cb, device,
-				   NULL, error)) {
-		g_prefix_error (error, "failed to write firmware: ");
+	if (flashrom_programmer_init(&flashprog, "internal", NULL)) {
+		g_error ("Error: Programmer initialization failed.");
```

Comment:
these all need to be `g_set_error ()` -- `g_error` in plugin code crashes the daemon.